src/bucket/BucketBase.cpp

```diff
@@ -306,7 +306,8 @@ BucketBase::merge(BucketManager& bucketManager, uint32_t maxProtocolVersion,
                   std::shared_ptr<BucketT> const& newBucket,
                   std::vector<std::shared_ptr<BucketT>> const& shadows,
                   bool keepTombstoneEntries, bool countMergeEvents,
-                  asio::io_context& ctx, bool doFsync)
+                  asio::io_context& ctx, bool doFsync,
+                  bool rewriteLiveToInitEntries)
```

Comment:
Sorry, I don't think I was clear in my original comment. I don't think we need another bool for `rewriteLiveToInitEntries`. This leads to footguns, as it should always be the case (currently) that when `keepTombstoneEntries == true`, `rewriteLiveToInitEntries == true`. When I was talking about the implicit assumption, I didn't mean to add a flag, as this adds more complexity for future maintainers. Rather, we should keep the interface as is, but add a test that will trip. Sorry about that.